File: accelerators/aks-sb-azmonitor-microservices/src/operations-api/src/main/java/com/microsoft/cse/cargoprocessing/operations/api/models/Cargo.java
package com.microsoft.cse.cargoprocessing.operations.api.models;

import java.io.Serializable;
import java.sql.Timestamp;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Data;

@Data
public class Cargo implements Serializable {
  private String id;
  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", timezone = "GMT")
  private Timestamp timestamp;
  private Product product;
  private Port port;
  private DemandDates demandDates;
  private Boolean valid;
  private String errorMessage;
}

File: accelerators/aks-sb-azmonitor-microservices/src/cargo-processing-validator/src/models/Cargo.ts
export interface Cargo {
  timestamp: Date;
  id: string;
  product: {
    name: string;
    quantity: number;
  };
  port: {
    source: string;
    destination: string;
  };
  demandDates: {
    start: Date;
    end: Date;
  };
}
